public/pages/QueryInsights/QueryInsights.tsx

```diff
@@ -88,56 +160,85 @@ const QueryInsights = ({
     {
       field: MEASUREMENTS_FIELD,
       name: LATENCY,
-      render: (measurements: any) => {
-        const latencyValue = measurements?.latency?.number;
-        return latencyValue !== undefined ? `${latencyValue} ms` : METRIC_DEFAULT_MSG;
+      render: (measurements: SearchQueryRecord['measurements']) => {
+        const result = calculateMetric(
+          measurements?.latency?.number,
+          measurements?.latency?.count,
+          1,
+          METRIC_DEFAULT_MSG
+        );
+        return `${result} ms`;
       },
       sortable: true,
       truncateText: true,
     },
     {
       field: MEASUREMENTS_FIELD,
       name: CPU_TIME,
-      render: (measurements: { cpu?: { number?: number } }) => {
-        const cpuValue = measurements?.cpu?.number;
-        return cpuValue !== undefined ? `${cpuValue / 1000000} ms` : METRIC_DEFAULT_MSG;
+      render: (measurements: SearchQueryRecord['measurements']) => {
+        const result = calculateMetric(
+          measurements?.cpu?.number,
+          measurements?.cpu?.count,
+          1000000, // Divide by 1,000,000 for CPU time
+          METRIC_DEFAULT_MSG
+        );
+        return `${result} ms`;
       },
       sortable: true,
       truncateText: true,
     },
     {
       field: MEASUREMENTS_FIELD,
       name: MEMORY_USAGE,
-      render: (measurements: { memory?: { number?: number } }) => {
-        const memoryValue = measurements?.memory?.number;
-        return memoryValue !== undefined ? `${memoryValue} B` : METRIC_DEFAULT_MSG;
+      render: (measurements: SearchQueryRecord['measurements']) => {
+        const result = calculateMetric(
+          measurements?.memory?.number,
+          measurements?.memory?.count,
+          1,
+          METRIC_DEFAULT_MSG
+        );
+        return `${result} B`;
       },
       sortable: true,
       truncateText: true,
     },
     {
       field: INDICES_FIELD,
       name: INDICES,
-      render: (indices: string[]) => Array.from(new Set(indices.flat())).join(', '),
+      render: (indices: string[], query: SearchQueryRecord) => {
+        const isSimilarity = query.group_by === 'SIMILARITY';
+        return isSimilarity ? '-' : Array.from(new Set(indices.flat())).join(', ');
```

Comment:
Why are we not showing the indices of the group here?

Reply:
The group only contains indices of the first query in the group. Since displaying this information might be misleading we decided to not display this here after consulting with UX.